src/script/components/HistoryExport/HistoryExport.tsx

```diff
@@ -179,15 +179,19 @@ const HistoryExport = ({switchContent, user, clientState = container.resolve(Cli
       setNumberOfRecords(numberOfRecords);
       setNumberOfProcessedRecords(0);
 
-      const archiveBlob = await backupRepository.generateHistory(
```

Comment:
Just added an undefined check, this was a hidden "Problem" before since the currentClient cut have been undefined

Reply:
In the future this should be injected, we should not consume the state so deep in the layers of the app (not a change for now :) )